docs/docs/recipes/transforming-data.md

```diff
@@ -54,3 +54,119 @@ export const query = graphql`
 
 - [Tutorial on transforming Markdown to HTML](/tutorial/part-six/#transformer-plugins) using `gatsby-transformer-remark`
 - Browse available transformer plugins in the [Gatsby plugin library](/plugins/?=transformer)
+
+## Transforming images into grayscale using GraphQL
+
+### Prerequisites
+
+- A [Gatsby site](/docs/quick-start) with `gatsby-config.js` and an `index.js` page
+- The `gatsby-image`, `gatsby-transformer-sharp`, and `gatsby-plugin-sharp` packages installed
+- A source plugin installed, such as `gatsby-source-filesystem`
+- An image (`.jpg`, `.png`, `.gif`, `.svg`, etc.) in the `src/images` folder
+
+### Directions
+
+1.
+
+Edit your gatsby-config.js file to source images and configure plugins for Gatsby's GraphQL data layer. A common approach is to source them from an images directory using the gatsby-source-filesystem plugin:
+
+```JavaScript:title=gatsby-config.js
+
+ plugins: [
+   {
+     resolve: `gatsby-source-filesystem`,
+     options: {
+       name: `images`,
+       path: `${__dirname}/src/images`,
+     },
+   },
+   `gatsby-transformer-sharp`,
+   `gatsby-plugin-sharp`,
+ ],
+```
+
+2. You would then need to create a query for your image that specifies all parameters that the gatsby image requires
+
+Note the `gatsby-source-filesystem` path pointing to the location of images in your project.
+
+Note: You can find these and other parameters in your GraphQL playground located at `http://localhost:8000/__graphql`
+
+```query {
+     file(relativePath: { eq: "gatsby-astronaut.png" }) {
+       childImageSharp {
+         // highlight-next-line
+         fluid(grayscale: true) {
+           ...GatsbyImageSharpFluid
+         }
+       }
+     }
+   }
+```
+
+Note: The relative path is the image path that you configured in `gatsby-source-filesystem`.
+
+```JavaScript
```

Comment:
I'm not sure this code snippet belongs here? We have the config in step 1 and the query in step 2. I believe this can be removed.